app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+  # require login needs to be created in the ApplicationController to create flash message
+  # skip_before_action :require_login, only: [:new, :create]
+  before_action :redirect_if_logged_in, only: [:new, :create]
+  def new
+
+  end
+
+  def create
+  data = params[:session_data]
+  @merchant = Merchant.find_by_email(data[:email])
+    if !@merchant.nil?
```

Comment:
Because Ruby is friendly and makes sense when you read it out loud, keeping that in mind...If you're looking at this statement and reading it out loud does this statement make the most sense? Instead of 'if not @merchant nil' which is a boolean expression is there a better way to express this question? You might find another way which might be more clear. Clear is always best.